data/templates/firewall/nftables.j2

```diff
@@ -7,86 +7,147 @@
 delete table ip vyos_filter
 {% endif %}
 table ip vyos_filter {
-    chain VYOS_FW_FORWARD {
-        type filter hook forward priority 0; policy accept;
-{% if state_policy is vyos_defined %}
-        jump VYOS_STATE_POLICY
-{% endif %}
-{% if interface is vyos_defined %}
-{%     for ifname, ifconf in interface.items() %}
-{%         if ifconf.in is vyos_defined and ifconf.in.name is vyos_defined %}
-        iifname {{ ifname }} counter jump NAME_{{ ifconf.in.name }}
-{%         endif %}
-{%         if ifconf.out is vyos_defined and ifconf.out.name is vyos_defined %}
-        oifname {{ ifname }} counter jump NAME_{{ ifconf.out.name }}
-{%         endif %}
-{%     endfor %}
-{% endif %}
-        jump VYOS_POST_FW
+{% if ip is vyos_defined %}
+{%     if ip.forward is vyos_defined %}
+{%         set ns = namespace(sets=[]) %}
+{%         for prior, conf in ip.forward.items() %}
+{%             set def_action = conf.default_action %}
+    chain VYOS_FORWARD_{{ prior }} {
+        type filter hook forward priority {{ prior }}; policy {{ def_action }};
+{%             if conf.rule is vyos_defined %}
+{%                 for rule_id, rule_conf in conf.rule.items() if rule_conf.disable is not vyos_defined %}
+        {{ rule_conf | nft_rule('FWD', prior, rule_id) }}
+{%                     if rule_conf.recent is vyos_defined %}
+{%                         set ns.sets = ns.sets + ['FWD_' + prior + '_' + rule_id] %}
+{%                     endif %}
+{%                 endfor %}
+{%             endif %}
     }
-    chain VYOS_FW_LOCAL {
-        type filter hook input priority 0; policy accept;
-{% if state_policy is vyos_defined %}
-        jump VYOS_STATE_POLICY
-{% endif %}
-{% if interface is vyos_defined %}
-{%     for ifname, ifconf in interface.items() %}
-{%         if ifconf.local is vyos_defined and ifconf.local.name is vyos_defined %}
-        iifname {{ ifname }} counter jump NAME_{{ ifconf.local.name }}
-{%         endif %}
-{%     endfor %}
-{% endif %}
-        jump VYOS_POST_FW
+{%         endfor %}
+{%         for set_name in ns.sets %}
+    set RECENT_{{ set_name }} {
```

Comment:
Do you think the namespace/sets object is better being in global scope and accumulated over the nodes, then iterated only once at the end?